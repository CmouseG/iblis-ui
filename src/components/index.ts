export { //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues
} from './form_components';
export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, //
    IblisTextField, IblisTextFieldProps, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ProgressBar, ProgressBarProps, //
    UploadAvatar, UploadAvatarProps
} from './ui_components';
