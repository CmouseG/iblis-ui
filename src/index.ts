export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    ConfirmSignupForm, ConfirmSignupFormProps, ConfirmSignupFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    ForceChangePasswordForm, ForceChangePasswordFormProps, ForceChangePasswordFormValues, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, IblisTextField, IblisTextFieldProps, //
    LoginForm, LoginFormProps, LoginFormValues, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    PasswordConfirmForm, PasswordConfirmFormProps, PasswordConfirmFormValues, //
    ProgressBar, ProgressBarProps, //
    ResetPasswordForm, ResetPasswordFormProps, ResetPasswordFormValues, //
    SignupForm, SignupFormProps, SignupFormValues, //
    StripeForm, StripeFormProps, //
    UploadAvatar, UploadAvatarProps, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues //
} from './components';
export { //
    CardMenu, CardMenuLevelProps, CardMenuProps, //
    CardTitleLayout, CardTitleLayoutProps, //
    ContentWithTitle, ContentWithTitleProps, //
    ExternalFormLayout, ExternalFormLayoutProps, //
    MainLayout, MainLayoutProps
} from './layouts';
export { theme } from './theme';
// tslint:disable-next-line:no-import-side-effect
import 'iblis-font/index.css';
// tslint:disable-next-line:no-import-side-effect
import 'typeface-roboto/index.css';
