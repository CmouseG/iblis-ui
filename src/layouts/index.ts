export { CardMenu, CardMenuLevelProps, CardMenuProps } from './card_menu';
export { CardTitleLayout, CardTitleLayoutProps } from './card_title_layout';
export { ContentWithTitle, ContentWithTitleProps } from './content_with_title';
export { ExternalFormLayout, ExternalFormLayoutProps } from './external_form_layout';
export { MainLayout, MainLayoutProps } from './main_layout';
