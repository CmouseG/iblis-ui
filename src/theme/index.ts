export { theme } from './iblis_theme';
