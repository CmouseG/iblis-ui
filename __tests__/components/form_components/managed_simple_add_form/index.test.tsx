import * as React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { combineReducers, createStore, ReducersMapObject, Store } from 'redux';
import { reducer as formReducer } from 'redux-form';
import { ManagedSimpleForm, ManagedSimpleFormProps } from '../../../../src';

it('renders correctly', () => {

    // Combine all app Reducers
    const reducersList: ReducersMapObject = {
        form: formReducer,
    };

    // Combine all app Reducers
    const reducers = combineReducers(reducersList);

    // create redux store with all app reducers amd all the middleware
    const store: Store<{}> = createStore(
        // reducers
        reducers,
        // preload state
        {},
    );

    const data = [
        { id: '1', name: 'Marketing' },
        { id: '2', name: 'Legal' },
    ];
    const ManagedSimpleFormOwnProps: ManagedSimpleFormProps = {
        data: data,
        idFieldName: 'id',
        displayFieldName: 'name',
        namePlaceHolder: 'Enter your name',
        addAction: (_value) => undefined,
        editAction: (_value) => undefined,
        deleteAction: (_id) => undefined,
        isLoading: false,
    };

    const tree = renderer
        .create(
            <Provider store={store}>
                <ManagedSimpleForm
                    {...ManagedSimpleFormOwnProps}
                />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
