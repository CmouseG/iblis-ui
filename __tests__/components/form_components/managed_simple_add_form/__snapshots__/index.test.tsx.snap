// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-container-1"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-8-36"
          style={
            Object {
              "paddingRight": 8,
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="MuiFormControl-root-99 IblisTextFieldBase-textFieldRoot-97 MuiFormControl-fullWidth-102"
              >
                <div
                  className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104"
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInput-input-111 IblisTextFieldBase-textFieldInput-98"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter your name"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <span
                className="MuiTypography-root-116 MuiTypography-caption-126 MuiTypography-colorError-139"
              >
                *
              </span>
            </div>
            <span
              className="MuiTypography-root-116 MuiTypography-caption-126 MuiTypography-colorError-139"
            >
              Â 
            </span>
          </div>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <button
            className="MuiButtonBase-root-164 MuiButton-root-140 MuiButton-contained-149 MuiButton-raised-152"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="submit"
          >
            <span
              className="MuiButton-label-141"
            >
              Add
            </span>
            <span
              className="MuiTouchRipple-root-167"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-flex-start-10"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-8-36 MuiGrid-grid-sm-10-52"
            style={
              Object {
                "height": 60,
              }
            }
          >
            <h3
              className="MuiTypography-root-116 MuiTypography-subheading-123 MuiTypography-noWrap-132"
              style={
                Object {
                  "paddingBottom": 8,
                  "paddingLeft": 9,
                  "paddingTop": 16,
                }
              }
            >
              Marketing
            </h3>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-4-32 MuiGrid-grid-sm-2-44"
            style={
              Object {
                "paddingRight": 8,
                "paddingTop": 4,
                "textAlign": "right",
              }
            }
          >
            <div>
              <button
                className="MuiButtonBase-root-164 MuiIconButton-root-174"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-179"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-180"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-167"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div>
    <div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-flex-start-10"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-8-36 MuiGrid-grid-sm-10-52"
            style={
              Object {
                "height": 60,
              }
            }
          >
            <h3
              className="MuiTypography-root-116 MuiTypography-subheading-123 MuiTypography-noWrap-132"
              style={
                Object {
                  "paddingBottom": 8,
                  "paddingLeft": 9,
                  "paddingTop": 16,
                }
              }
            >
              Legal
            </h3>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-4-32 MuiGrid-grid-sm-2-44"
            style={
              Object {
                "paddingRight": 8,
                "paddingTop": 4,
                "textAlign": "right",
              }
            }
          >
            <div>
              <button
                className="MuiButtonBase-root-164 MuiIconButton-root-174"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-179"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-180"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-167"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
